tests/io/lobster/test_outputs.py

```diff
@@ -1603,7 +1603,7 @@ def test_has_good_quality(self):
         assert not self.band_overlaps2_new.has_good_quality_check_occupied_bands(
             number_occ_bands_spin_up=10, limit_deviation=0.0000001
         )
-        assert not self.band_overlaps2.has_good_quality_check_occupied_bands(
+        assert self.band_overlaps2.has_good_quality_check_occupied_bands(
```

Comment:
So, the result will change from false to true?

Reply:
Yes, as the logic was not correct before, the results will change.

Reply:
Hi @JaGeo , refactored the code a bit and also added a test to ensure we extract and check correct data